service/src/test/java/bio/terra/cbas/controllers/TestMethodsApiController.java

```diff
@@ -99,17 +97,20 @@ private void initSamMocks() {
   private void initMocks() {
     initSamMocks();
 
-    when(methodDao.getMethods()).thenReturn(List.of(neverRunMethod1, previouslyRunMethod2));
-    when(methodDao.getMethod(neverRunMethod1.methodId())).thenReturn(neverRunMethod1);
-    when(methodDao.getMethod(previouslyRunMethod2.methodId())).thenReturn(previouslyRunMethod2);
-    when(githubMethodDetailsDao.getMethodSourceDetails(previouslyRunMethod2.methodId()))
-        .thenReturn(githubMethodDetails);
-    when(githubMethodDetailsDao.getMethodSourceDetails(neverRunMethod1.methodId()))
-        .thenReturn(githubMethodDetails);
+    Method neverRunMethod1WithGithub = neverRunMethod1.withGithubMethodDetails(githubMethodDetails);
+    Method previouslyRunMethod2WithGithub =
```

Comment:
Not necessarily something to fix in this PR... this amount of mocking in the API just to check whether a single function gets called in a particular way feels like a code smell.

I don't think it's a new through that perhaps our controller methods would benefit from being divided with much of the logic pushed down into smaller functions in the services layer!

Reply:
Yeah, I agree... maybe I can find some opportunity here to chip away at this controller layer division...